### PR TITLE
Fix qwen eagle3 precision issue

### PR DESCRIPTION
**Pull request by zichen-kuuga**
The bug in vllm-musa/vllm_musa/v1/attention/backends/flash_attn.py has been fixed
  - The root cause is that musa FlashAttention separates decode/prefill for processing, with speculative verifier the multi-query decode using flash_attn_with_kvcache results in incorrect logits.
  - Now it has been changed to uniformly use the MATE flash_attn_varlen_func path, in line with upstream the spec-decode attention of vLLM is semantically consistent.

Test command: vllm serve /home/dist/Qwen3-8B -ac.backend FLASH_ATTN --cudagraph-capture-sizes 4 --speculative-config '{"method": "eagle3", "model": "/home/dist/Qwen3-8B_eagle3", "num_speculative_tokens":
 3}'
80d6321720414c9d993487de4a241b26

### CODE REVIEW

File: vllm_musa/v1/attention/backends/flash_attn.py [hunk 1]
```diff
@@ -828,72 +827,37 @@ def forward(
...
-                    # MATE returns output already flattened: [num_tokens, num_heads * head_size]
-                    # Just copy directly without reshaping
-                    output[prefill_start:prefill_end] = prefill_output
+                # Use MATE's varlen paged-cache path for target-model attention.
+                # Speculative decoding can send more than one query token per
+                # request (bonus + draft verification tokens), so splitting all
+                # decode requests through flash_attn_with_kvcache corrupts the
+                # verifier logits. The varlen path matches vLLM upstream and
+                # handles query_start_loc/max_query_len directly.
+                flash_attn_varlen_func(
```
[the remark this answers is not in the document]
Comment (zichen-kuuga) [reply vllm_musa/v1/attention/backends/flash_attn.py:836]: Support out parameters, no need to display the use of return values

**Review by yeahdongcn — APPROVED** [review]
Please add some logs for further debugging/reference.

**Comment by zichen-kuuga** [comment]
> Please add some logs for further debugging/reference.

done
